### PR TITLE
(1249) Add BEIS ID and export it to the report csv

### PR DESCRIPTION
**Pull request by mec**
## Changes in this PR

To aid BEIS users in the period whilst the service is introduced and
runs along with the tracker spreadsheets we should capture and export
what is known as the BEIS ID.

This is an identifier used inside BEIS to track a single activity. Some
Delivery partner identifiers will use the BEIS ID and some will not, so
we should keep it separately to the delivery partner identifier to cover
all instances.

This field will not be visible in the application at all, it is simply
'passed through' to the report csv so up-stream systems can match a RODA
ID to an existing BEIS ID, with the goal of replacing BEIS IDs with RODA
IDs long term.

This fields value will only ever be set by historic data imported to the
service by the team during delivery partner on-boarding, never from an
end user.

Once the migration of all delivery partners to RODA is complete, this
field becomes redundant, but could remain and simply no longer appear in
the report csv.

The choice to neither restrict or validate this field is intentional as we
cannot guarantee the uniqueness of these IDs or their format and some
activities may not have one at all.

The work to import values into this column does not form part of this PR 
and will follow later.

## Next steps

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has no impact outside the development team.
- [ ] Do any environment variables need amending or adding?
- [ ] Have any changes to the XML been checked with the IATI validator? See [XML Validation](https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/develop/doc/xml-validation.md)

### CODE REVIEW

**Comment by mec** [comment]
[Capture and store the 'BEIS ID' to facilitate BEIS migration to using the RODA ID](https://trello.com/c/GzvOm347/1249-capture-and-store-the-beis-id-to-facilitate-beis-migration-to-using-the-roda-id)

File: app/presenters/activity_csv_presenter.rb [hunk 1]
```diff
@@ -3,4 +3,9 @@ def intended_beneficiaries
     return if super.blank?
     to_model.intended_beneficiaries.map { |item| I18n.t("activity.recipient_country.#{item}") }.join("; ")
   end
+
+  def beis_id
+    return "" if super.blank?
```
Comment (pezholio) [inline app/presenters/activity_csv_presenter.rb:8]: Do we need to do this? `nil.to_s` is functionally the same as doing this

**Review by pezholio — APPROVED** [review]
Looks good, just one question on the presenter, but not a blocker

File: app/presenters/activity_csv_presenter.rb (hunk 1 above)
Comment (mec) [reply app/presenters/activity_csv_presenter.rb]: 🤦‍♀️ yes, thanks for pointing this out! 😄

**Comment by jcoglan** [comment]
Where did the docs for identifiers go? I have my [gist][1] but I don't recall where this got incorporated into the service's docs. It would be worth adding this new identifier to that document and sharing it; I've been pointing people to the gist in order to explain IDs and that doesn't feel sustainable in terms of maintaining the document.

[1]: https://gist.github.com/jcoglan/e525c9059e38f0c812069f6171c37652

**Pull request merged**